kafka_consumer_freshness_tracker/src/test/java/com/tesla/data/consumer/freshness/ConsumerFreshnessTest.java

```diff
@@ -272,6 +277,156 @@ public void testBurrowMissingConsumerGroupPartitionEndOffsetMarksMissing() throw
+      freshness.setupWithBurrow(conf, burrow);
+
+      assertEquals("Should be no worker queue for invalid cluster", 0, freshness.getAvailableWorkersForTesting().size(),
+          0.0);
+    });
+  }
+
+  @Test
+  public void testStrictModeConfigurationIsValidServersMatchBurrow() throws Exception {
+    Burrow burrow = mock(Burrow.class);
+    String clusterName = "cluster1";
+    when(burrow.getClusterDetail(clusterName))
+      .thenReturn(mockBurrowClusterDetailResponse(
+            Arrays.asList("kafka01.example.com:10251", "kafka02.example.com:10251", "kafka03.example.com:10251")
+            )
+          );
+
+    Map<String, Object> conf = mockConfForCluster(
+        clusterName,
+        "kafka01.example.com:10251, kafka02.example.com:10251, kafka03.example.com:10251"
+        );
+
+    withExecutor(executor -> {
+      ConsumerFreshness freshness = new ConsumerFreshness(true);
+
+      // This exception should be thrown when the consumers are attempted to be
+      // connected after validation succeeds
+      thrown.expect(KafkaException.class);
+      thrown.expectMessage("Failed to construct kafka consumer");
+
+      freshness.setupWithBurrow(conf, burrow);
+
+      assertEquals("Should be a worker queue for valid cluster", 1, freshness.getAvailableWorkersForTesting().size(),
+          0.0);
+    });
+  }
+
+  @Test
+  public void testStrictmodeConfigurationIsInvalidMissingServer() throws Exception {
+    Burrow burrow = mock(Burrow.class);
+    String clusterName = "cluster1";
+    when(burrow.getClusterDetail(clusterName))
+      .thenReturn(mockBurrowClusterDetailResponse(
+            Arrays.asList("kafka01.example.com:10251", "kafka02.example.com:10251",
+              "kafka03.example.com:10251", "kafka04.example.com:10251")
+            ));
+
+    Map<String, Object> conf = mockConfForCluster(
+        clusterName,
+        "kafka01.example.com:10251, kafka02.example.com:10251, kafka03.example.com:10251"
+        );
+
+    withExecutor(executor -> {
+      ConsumerFreshness freshness = new ConsumerFreshness(true);
+      freshness.setupWithBurrow(conf, burrow);
+
+      assertEquals("Should be no worker queue for invalid cluster", 0, freshness.getAvailableWorkersForTesting().size(),
+          0.0);
+    });
+  }
+
+  Map<String, Object> mockConfForCluster(String name, String bootstrapServers) {
+    Map<String, Object> conf = new HashMap<String, Object>();
+    List<Map<String, Object>> clusterList = new LinkedList<Map<String, Object>>();
+    Map<String, Object> clusterConf = new HashMap<String, Object>();
+    Map<String, Object> kafkaConf = new HashMap<String, Object>();
+    kafkaConf.put("bootstrap.servers", bootstrapServers);
+    clusterConf.put("name", name);
+    clusterConf.put("kafka", kafkaConf);
+    clusterList.add(clusterConf);
+    conf.put("clusters", clusterList);
+    return conf;
+  }
+
+  Map<String, Object> mockBurrowClusterDetailResponse(List<String> servers) {
```

Comment:
A varargs here would make the callers slightly less verbose but not lose clarity

Reply:
done